src/main.cpp

```diff
@@ -5654,24 +5654,26 @@ bool SendMessages(CNode* pto)
         //
         // Message: getdata (non-blocks)
         //
-        while (!pto->fDisconnect && !pto->mapAskFor.empty() && (*pto->mapAskFor.begin()).first <= nNow)
+        if(!IsInitialBlockDownload())
```

Comment:
I think the guard on line 4566 is sufficient, from my reading of the code and a quick test.  Is there a case I'm missing?